src/index.ts

```diff
@@ -1,102 +1,224 @@
+            /[/\\/?%*:|"<>]/g,
+            "-",
+        )}_digest.txt`;
     }
 
-    const args = [ 'push',
-        '--quiet',
-        '--digestfile', digestFile,
-        '--creds', creds,
+    // push the image
+    const args = [
+        "push",
+        "--quiet",
+        "--digestfile",
+        digestFile,
+        "--creds",
+        creds,
         imageToPush,
-        registryPath
+        registryPath,
     ];
 
     // check if tls-verify is not set to null
     if (tlsVerify) {
         args.push(`--tls-verify=${tlsVerify}`);
     }
 
-    await execute(podman, args);
+    await execute(await getPodmanPath(), args);
 
     core.info(`Successfully pushed ${imageToPush} to ${registryPath}.`);
-    core.setOutput('registry-path', registryPath);
+    core.setOutput("registry-path", registryPath);
+
+    // remove the pulled image from the Podman image storage
+    if (isPushingDockerImage) {
+        core.info(`Removing ${imageToPush} from the Podman image storage`);
+        await execute(await getPodmanPath(), [ "rmi", imageToPush ]);
+    }
 
     try {
         const digest = (await fs.promises.readFile(digestFile)).toString();
         core.info(digest);
-        core.setOutput('digest', digest);
+        core.setOutput("digest", digest);
     }
     catch (err) {
         core.warning(`Failed to read digest file "${digestFile}": ${err}`);
     }
 }
 
-async function execute(executable: string, args: string[], execOptions: exec.ExecOptions = {}): Promise<{ exitCode: number, stdout: string, stderr: string }> {
+async function pullImageFromDocker(
+    imageName: string,
+): Promise<boolean> {
+    try {
+        await execute(await getPodmanPath(), [ "pull", `docker-daemon:${imageName}` ]);
+        core.info(`Image ${imageName} found in Docker image storage`);
+        return true;
+    }
+    catch (err) {
+        core.info(`Image ${imageName} not found in Docker image storage`);
+        return false;
+    }
+}
+
+async function checkImageInPodman(
+    imageName: string,
+): Promise<boolean> {
+    // check if images exist in Podman's storage
+    core.info("Checking image in Podman image storage");
+    try {
+        await execute(await getPodmanPath(), [ "image", "exists", imageName ]);
+        core.info(`Image ${imageName} found in Podman image storage`);
+        return true;
+    }
+    catch (err) {
+        core.info(`Image ${imageName} not found in Podman image storage`);
+        core.debug(err);
+        return false;
+    }
+}
+
+async function isPodmanLocalImageLatest(
+    imageName: string,
+): Promise<boolean> {
+    // get creation time of the image present in the Podman image storage
+    const podmanLocalImageTimeStamp = await execute(await getPodmanPath(), [
+        "image",
+        "inspect",
+        imageName,
+        "--format",
+        "{{.Created}}",
+    ]);
+
+    core.info(`The image from Podman image storage was last modified at ${podmanLocalImageTimeStamp.stdout}`);
```

Comment:
Sorry, it didn't occur to me that the stdout is already echoed, so now it's there twice. You can remove this and the other one below.

![image](https://user-images.githubusercontent.com/8888438/104958662-6fc7c500-599e-11eb-92a3-a7078aa425b2.png)